list.go

```diff
@@ -0,0 +1,91 @@
+package pack
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"reflect"
+)
+
+type List struct {
+	T     Type
+	Elems []Value
+}
+
+func NewList(vs ...interface{}) List {
+	elems := make([]Value, len(vs))
+	for i := range elems {
+		elems[i] = vs[i].(Value)
+	}
+	return List{
+		T:     elems[0].Type(),
+		Elems: elems,
+	}
+}
+
+// Type returns the list type.
+func (v List) Type() Type {
+	return typeList{
+		Type: v.T,
+		Size: uint64(len(v.Elems)),
+	}
+}
+
+// SizeHint returns the number of bytes required to represent the list in
+// binary.
+func (v List) SizeHint() int {
+	return v.T.SizeHint() * len(v.Elems)
```

Comment:
```suggestion
	return surge.SizeHint(v.Elems)
```
Values do not marshal type information.